src/Groups/GrpAb.jl

```diff
@@ -12,3 +12,348 @@ function restrict_codomain(f::GrpAbFinGenMap)
+
+function representative_action(G::GrpAbFinGen, x::GrpAbFinGenElem, y::GrpAbFinGenElem)
+   x == y ? (true, zero(G)) : (false, nothing)
+end
+
+
+################################################################################
+#
+#   Conjugacy classes of subgroups
+#
+################################################################################
+
+conjugacy_class(G::T, H::T) where T <: GrpAbFinGen = GrpAbFinGenConjClass(G, H)
+
+function conjugacy_classes_subgroups(G::GrpAbFinGen)
+   return [conjugacy_class(G, H) for (H, mp) in subgroups(G)]
+end
+
+#TODO: Does Hecke provide the following?
+# subgroup_reps(G::GrpAbFinGen; order::ZZRingElem = ZZRingElem(-1))
+# conjugacy_classes_maximal_subgroups(G::GrpAbFinGen)
+# maximal_subgroup_reps(G::GrpAbFinGen)
+# low_index_subgroup_reps(G::GrpAbFinGen, n::Int)
+
+conjugate_group(G::GrpAbFinGen, x::GrpAbFinGenElem) = G
+
+Base.:^(G::GrpAbFinGen, x::GrpAbFinGenElem) = G
+
+function is_conjugate(G::GrpAbFinGen, H::GrpAbFinGen, K::GrpAbFinGen)
+   if is_subgroup(H, K)[1] && is_subgroup(K, H)[1]
+     return true
+   else
+     return false
+   end
+end
+
+function representative_action(G::GrpAbFinGen, H::GrpAbFinGen, K::GrpAbFinGen)
+   if is_subgroup(H, K)[1] && is_subgroup(K, H)[1]
+     return true, zero(G)
+   else
+     return false, nothing
+   end
+end
+
+is_conjugate_subgroup(G::T, U::T, V::T) where T <: GrpAbFinGen = is_subgroup(V, U)[1]
+
+
+Base.IteratorSize(::Type{<:GrpAbFinGenConjClass}) = Base.HasLength()
+
+Base.iterate(C::GrpAbFinGenConjClass) = iterate(C, 0)
+
+function Base.iterate(C::GrpAbFinGenConjClass, state::Int)
+  if state == 0
+    return representative(C), 1
+  else
+    return nothing
+  end
+end
+
+
+################################################################################
+#
+# Normal Structure
+#
+################################################################################
+
+function core(G::GrpAbFinGen, H::GrpAbFinGen)
+  flag, emb = is_subgroup(H, G)
+  flag || error("H  must be a subgroup of G")
+  return (H, emb)
+end
+
+function normalizer(G::GrpAbFinGen, H::GrpAbFinGen)
+  is_subgroup(H, G)[1] || error("H  must be a subgroup of G")
+  return (G, identity_map(G))
+end
+
+function normalizer(G::GrpAbFinGen, x::GrpAbFinGenElem)
+  is_element(x, G) || throw(ArgumentError("x does not lie in G"))
+  return (G, identity_map(G))
+end
+
+function normal_closure(G::GrpAbFinGen, H::GrpAbFinGen)
+  flag, emb = is_subgroup(H, G)
+  flag || error("H  must be a subgroup of G")
+  return (H, emb)
+end
+
+pcore(G::GrpAbFinGen, p::IntegerUnion) = sylow_subgroup(G, p)
+
+
+################################################################################
+#
+# Specific Subgroups
+#
+################################################################################
+
+fitting_subgroup(G::GrpAbFinGen) = (G, identity_map(G))
+
+function frattini_subgroup(G::GrpAbFinGen)
+   is_finite(G) || throw(ArgumentError("G is not finite"))
```

Comment:
Could also use the `@req` macro:
```suggestion
   @req is_finite(G) "G is not finite"
```